### PR TITLE
Disallow magic numbers

### PR DESCRIPTION
**Pull request by benedikt-richter**
### Summary of changes

In order to improve readability of our code, magic numbers are now forbidden by the linter. Current errors are fixed. Many are just ignored and have to be fixed successively.

### Context and reason for change

Improve readability of our code.

### How can the changes be tested

`yarn lint`

### CODE REVIEW

File: src/Frontend/Components/AttributionWizardPopup/attribution-wizard-popup-helpers.tsx [hunk 1]
```diff
@@ -312,7 +312,8 @@ function getWizardListItem(
 }
 
 function getCountText(count: number, totalAttributeCount: number): string {
-  const percentage = (count / totalAttributeCount) * 100;
+  const percentageFactor = 100;
+  const percentage = (count / totalAttributeCount) * percentageFactor;
```
Comment (MarkusObendrauf) [inline src/Frontend/Components/AttributionWizardPopup/attribution-wizard-popup-helpers.tsx:316]: would be in favour of allowing `100`

File: src/Frontend/Components/BackendCommunication/__tests__/BackendCommunication.test.tsx [hunk 2]
```diff
@@ -16,7 +16,8 @@ import { Attributions, ExportType } from '../../../../shared/shared-types';
 describe('BackendCommunication', () => {
   it('renders an Open file icon', () => {
     renderComponentWithStore(<BackendCommunication />);
-    expect(window.electronAPI.on).toHaveBeenCalledTimes(11);
+    const expectedNumberOfCalls = 11;
+    expect(window.electronAPI.on).toHaveBeenCalledTimes(expectedNumberOfCalls);
```
Comment (MarkusObendrauf) [inline src/Frontend/Components/BackendCommunication/__tests__/BackendCommunication.test.tsx:20]: extracting like this in an assert just hurts readability imo

**Review by lennartclaas — APPROVED** [review]
I agree that prohibiting magic numbers in tests feels somewhat odd. But I have no strong opinion.

File: src/Frontend/Components/AttributionWizardPopup/attribution-wizard-popup-helpers.tsx (hunk 1 above)
Comment (benedikt-richter) [reply src/Frontend/Components/AttributionWizardPopup/attribution-wizard-popup-helpers.tsx:316]: That's fine with me.

File: src/Frontend/Components/BackendCommunication/__tests__/BackendCommunication.test.tsx (hunk 2 above)
Comment (benedikt-richter) [reply src/Frontend/Components/BackendCommunication/__tests__/BackendCommunication.test.tsx:20]: I think there is no reasonable way to disable a single lint rule for test files. 
In my opinion, it does also increase readablility here. However, would be fine with ignoring the lint in cases where it makes sense (like some in this diff).

Comment (MarkusObendrauf) [reply src/Frontend/Components/BackendCommunication/__tests__/BackendCommunication.test.tsx:20]: the "before" version reads just like a sentence: "Expect window.electronAPI.on to have been called 11 times".
in the "after" version, you have to hold another value in your head as you're reading the code, and the code reads as "Expect window.electronAPI.on to have been called expected number of calls times". thats definitely harder to read.

**Comment by MarkusObendrauf** [comment]
My thoughts are: There are some places where this has a huge benefit, and some where its a small detriment. So overall, I think it's a good change. I would just like to mitigate as much weird stuff as possible (see review comments). It might be nice to get more opinions.

**Review by leslielazzarino — APPROVED** [review]

**Pull request merged**